src/components/MapComponent.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useRef, useState, useCallback } from 'react';
+import { GoogleMap, LoadScript } from '@react-google-maps/api';
+
+const containerStyle = {
+  width: '100%',
+  height: '400px',
+};
+
+interface MapComponentProps {
+  onLocationUpdate: (latitude: number, longitude: number) => void;
+}
+
+const MapComponent: React.FC<MapComponentProps> = ({ onLocationUpdate }) => {
+  const [latitude, setLatitude] = useState<number | null>(null);
+  const [longitude, setLongitude] = useState<number | null>(null);
+  const [callCount, setCallCount] = useState<number>(0);
+
+  // eslint-disable-next-line no-undef
+  const intervalRef = useRef<NodeJS.Timeout | null>(null);
+
+  // Control flag to limit API calls in development
+  const isDevelopment = process.env.NODE_ENV === 'development';
+
+  //const apiKey = process.env.REACT_APP_GOOGLE_MAPS_API_KEY; // this was my inital approch but its not working as expected
+  const apiKey = ''; //Comment when developing
+  //const apiKey = "AIzaSyCAwTonwIQ7T8AqFXgxk7ksCxol38ChLS8"; // uncomment when in developemnt
+
+  const fetchLocation = useCallback(() => {
+    if (
+      navigator.geolocation &&
+      (!isDevelopment || (isDevelopment && callCount < 10))
+    ) {
+      navigator.geolocation.getCurrentPosition(
+        (position) => {
+          const { latitude, longitude } = position.coords;
+          setLatitude(latitude);
+          setLongitude(longitude);
+          onLocationUpdate(latitude, longitude);
+          if (isDevelopment) {
+            setCallCount((prevCount) => prevCount + 1);
+          }
+        },
+        (error) => {
+          console.error('Error fetching geolocation:', error);
+        },
+        {
+          enableHighAccuracy: true,
+          timeout: 5000,
+          maximumAge: 0,
+        }
+      );
+    }
+
+    if (isDevelopment && callCount >= 10 && intervalRef.current) {
+      clearInterval(intervalRef.current);
+    }
+  }, [callCount, isDevelopment, onLocationUpdate]);
+
+  useEffect(() => {
+    intervalRef.current = setInterval(fetchLocation, 10000); // Fetch location every 10 seconds
+
+    return () => {
+      if (intervalRef.current) {
+        clearInterval(intervalRef.current);
+      }
+    };
+  }, [fetchLocation]);
```

Comment:
while using useEffect will work and not be buggy or affect performance, there will be a bug where setting longitude and latitude will be behind slightly. This is because useEffect will only run asynchronously after the render, affecting the accuracy of the location slightly. 

A good alternative in this case would be useMemo, but if its too much trouble just stick with useEffect tbh

Reply:
I think  this is fine for now, there is a warning and a recommended approach provided by google using await and Asynchronous calls to the map API, so if we do need to we can make a ticket to fix that warning when it poses a problem
![image](https://github.com/malabudi/deer-tracker-fe/assets/85903772/8e3de748-9349-4e45-8a04-b1c4f8dfb1ae)